[intersection-observer] Import WPT /intersection-observer

tuankiet65 opened the pull request:
#### 36ef1fe153fe86fc3c79194a72e02906affcc8eb
<pre>
[intersection-observer] Import WPT /intersection-observer
<a href="https://rdar.apple.com/165115590">rdar://165115590</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=302850">https://bugs.webkit.org/show_bug.cgi?id=302850</a>

Reviewed by Simon Fraser.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/c3e2c704fc37bb6e69c6d2b646b775f4e97c4991">https://github.com/web-platform-tests/wpt/commit/c3e2c704fc37bb6e69c6d2b646b775f4e97c4991</a>

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/animating-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/animating.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/clip-path-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/clip-path-animation.html:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/cross-origin-tall-iframe-rendering-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/cross-origin-tall-iframe-rendering.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/cross-origin-tall-iframe-rendering.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/cross-origin-tall-iframe.sub-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/cross-origin-tall-iframe.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/intersection-ratio-ib-split-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/intersection-ratio-ib-split.html:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/reinsert-element-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/reinsert-element.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/resources/cross-origin-tall-subframe.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/target-in-detached-document-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/target-in-detached-document.html:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/v2/animated-opacity-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/v2/animated-opacity.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/v2/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/w3c-import.log:

Canonical link: <a href="https://commits.webkit.org/303445@main">https://commits.webkit.org/303445@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3a05e9cef9b3489f971c66d30790f9bdb00fe830

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132406 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4901 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43448 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139921 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/84363 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/ed6fdf25-6853-49dd-9afd-1490568a09f1) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4914 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4660 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101221 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/68477 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/d600165e-9cc7-402a-9c57-7d448fb32de9) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135352 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/3447 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118598 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82013 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/20aa1526-4b55-4008-b7a1-6e9d76db3812) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/3330 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/1202 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/83147 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/112177 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/36715 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142573 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4571 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37304 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109597 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4653 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3941 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109776 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27823 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3461 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114872 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/57847 "Built successfully") | | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4625 "Build is in progress. Recent messages:OS: Tahoe (26.0), Xcode: 26.0.1; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33228 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4457 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4716 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4582 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

smfr reviewed (commented):
Make sure you garden for the failing/flakey tests.

tuankiet65 committed:
[intersection-observer] Import WPT /intersection-observer
rdar://165115590
https://bugs.webkit.org/show_bug.cgi?id=302850

Reviewed by Simon Fraser.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/c3e2c704fc37bb6e69c6d2b646b775f4e97c4991

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/animating-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/animating.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/clip-path-animation-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/clip-path-animation.html:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/cross-origin-tall-iframe-rendering-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/cross-origin-tall-iframe-rendering.sub-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/cross-origin-tall-iframe-rendering.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/cross-origin-tall-iframe.sub-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/cross-origin-tall-iframe.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/intersection-ratio-ib-split-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/intersection-ratio-ib-split.html:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/reinsert-element-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/reinsert-element.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/resources/cross-origin-tall-subframe.sub.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/resources/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/target-in-detached-document-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/target-in-detached-document.html:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/v2/animated-opacity-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/v2/animated-opacity.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/v2/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/intersection-observer/w3c-import.log:

Canonical link: https://commits.webkit.org/303445@main
